File: 1894-merge-strings-alternately/merge-strings-alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        result = ''
        i = 0
        n1 = len(word1)
        n2 = len(word2)
        while i < n1 and i < n2:
            result += word1[i] + word2[i]
            i += 1

        result += word1[i:]
        result += word2[i:]
        
        return result
        

File: 1894-merge-strings-alternately/merge-strings-alternately.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        int w1=0;
        int w2=0;
        int n1 = word1.size(), n2 = word2.size();
        string result = "";
        
        while(w1 < n1 or w2 < n2) {
            if(w1 < n1) {
                result += word1[w1++];
            }
            if(w2 < n2) {
                result += word2[w2++];
            }
            
        }

        return result;

    }
};